middleware/plugin/middleware.go

```diff
@@ -0,0 +1,140 @@
+//go:build plugin
+
+package plugin
+
+/*
+ * SPDX-FileCopyrightText: 2023 Siemens AG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Author: Michael Adler <michael.adler@siemens.com>
+ */
+
+import (
+	"net/http"
+	"net/url"
+
+	"github.com/Southclaws/fault"
+	"github.com/rs/zerolog/log"
+	genPlugin "github.com/siemens/wfx/generated/plugin"
+	"github.com/siemens/wfx/generated/plugin/client"
+	"github.com/siemens/wfx/middleware/logging"
+)
+
+type MW struct {
+	plugin Plugin
+}
+
+func NewMiddleware(plugin Plugin) (*MW, error) {
+	log.Debug().Str("plugin", plugin.Name()).Msg("Creating new plugin middleware")
+
+	if err := plugin.Start(); err != nil {
+		_ = plugin.Wait()
+		log.Err(err).Msg("Failed to start plugin")
+		return nil, fault.Wrap(err)
+	}
+	mw := MW{plugin: plugin}
+
+	go func() {
+		for {
+			p := mw.plugin
+
+			err := p.Wait()
+			log.Warn().Err(err).Msg("Plugin stopped")
+			if p.IsStopped() {
+				log.Info().Msg("Plugin was stopped, not restarting it")
```

Comment:
Hm, the question is whether this is appropriate? If there's a plugin, it should run at all times and be routed any request through, otherwise it's "complimentary". That, however, is not useful for authorization plugins?

Reply:
Yes, the name of the method is misleading. The actual semantics is that the variable is true if wfx was instructed to shut down; then, the idea is not to restart it. I have renamed it.